mono/utils/mono-threads-posix.c

```diff
@@ -35,6 +35,8 @@ extern int tkill (pid_t tid, int signal);
 
 #include <sys/resource.h>
 
+/* FIXME: Haiku lacks pthread_attr_setschedpolicy */
+#if !defined(__HAIKU__)
```

Comment:
The appropriate fix around setsched is the following:

- rename all related functions around mono_threads_platform_* (so mono_threads_platform_reset_priority here).
- in mono-threads.h add an ifdef for HAIKU and define MONO_THREADS_PLATFORM_HAS_SETSCHED
- in mono-threads-posix.c convert include the posix impl to check for the above define
- in mono-threads-haiku.c add a bunch of empty stubs.